sweetest/src/test/java/com/mysugr/sweetest/ProvideTest.kt

```diff
@@ -0,0 +1,189 @@
+package com.mysugr.sweetest
+
+import com.mysugr.sweetest.framework.base.BaseJUnitTest
+import com.mysugr.sweetest.framework.base.BaseSteps
+import com.mysugr.sweetest.framework.base.dependency
+import com.mysugr.sweetest.framework.base.steps
+import com.mysugr.sweetest.framework.configuration.ModuleTestingConfiguration
+import com.mysugr.sweetest.framework.configuration.moduleTestingConfiguration
+import com.mysugr.sweetest.framework.context.TestContext
+import com.mysugr.sweetest.framework.environment.TestEnvironment
+import com.mysugr.sweetest.util.mock
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Test
+import org.mockito.Mockito.`when`
+
+class ProvideTest {
```

Comment:
for simplicity i thing you could have merged steps and tests together into one, because under the hood they use the same accessor anyway, but no strong opinion. feel free to have a look at [my test](https://github.com/mysugr/sweetest/pull/28/files#diff-ed85968250e216f3c9fb5e3bd4210dc2) though...